robyn/__init__.py

```diff
@@ -211,11 +222,14 @@ def start(self, host: str = "127.0.0.1", port: int = 8080, _check_port: bool = T
         port = int(os.getenv("ROBYN_PORT", port))
         open_browser = bool(os.getenv("ROBYN_BROWSER_OPEN", self.config.open_browser))
 
+        valid_socket, _socket = self.get_socket(host, port)
+
         if _check_port:
-            while self.is_port_in_use(port):
-                logger.error("Port %s is already in use. Please use a different port.", port)
+            while not valid_socket:
                 try:
                     port = int(input("Enter a different port: "))
+
+                    valid_socket, _socket = self.get_socket(host, port)
```

Comment:
@VishnuSanal , let's not use `_socket` since it is being referenced. We pre prend `_` to show something not being used. Let's call it `socket`

Reply:
IIRC, I got a name shadowing warning.

Reply:
@VishnuSanal , but you are using the variable in the lines below

Reply:
my bad then, done! :)